### PR TITLE
add getting seagrass + soulsoil

### PR DESCRIPTION
**Pull request by remplerus**

### CODE REVIEW

**Comment by remplerus** [comment]
seagrass from sand at level iron, and soulsoil by adding coarse dirt to witchwater in a barrel (so coarse dirt has more options now :D )

**Review by NovaMachina — CHANGES_REQUESTED** [review]
Please include these recipes in the data generators.

File: src/main/java/novamachina/exnihilosequentia/common/datagen/ExNihiloRecipeGenerator.java [hunk 1]
```diff
@@ -576,25 +578,34 @@ private void registerMisc(Consumer<IFinishedRecipe> consumer) {
...
         CookingRecipeBuilder.cookingRecipe(Ingredient
                 .fromItems(EnumResource.SILKWORM.getRegistryObject().get()), ExNihiloItems.COOKED_SILKWORM.get(), 0.1F, 100, IRecipeSerializer.SMOKING)
                 .addCriterion("has_silkworm", InventoryChangeTrigger.Instance
                         .forItems(EnumResource.SILKWORM.getRegistryObject().get()))
                 .build(consumer, createSaveLocation(new ResourceLocation(ExNihiloConstants.ModIds.EX_NIHILO_SEQUENTIA, "cooked_silkworm_from_smoker")));
+        CookingRecipeBuilder.blastingRecipe(Ingredient
+                .fromItems(EnumResource.SILKWORM.getRegistryObject().get()), ExNihiloItems.COOKED_SILKWORM.get(), 0.1F, 100)
+                .addCriterion("has_silkworm", InventoryChangeTrigger.Instance
+                        .forItems(EnumResource.SILKWORM.getRegistryObject().get()))
+                .build(consumer, createSaveLocation(new ResourceLocation(ExNihiloConstants.ModIds.EX_NIHILO_SEQUENTIA, "cooked_silkworm_from_smoker")));
```
Comment (NovaMachina) [inline src/main/java/novamachina/exnihilosequentia/common/datagen/ExNihiloRecipeGenerator.java]: Duplicate recipe name

File: src/main/java/novamachina/exnihilosequentia/common/datagen/ExNihiloRecipeGenerator.java [hunk 2]
```diff
@@ -576,25 +578,34 @@ private void registerMisc(Consumer<IFinishedRecipe> consumer) {
...
                         .get(), 0.7F, 200)
                 .addCriterion("has_uncooked_crucible", InventoryChangeTrigger.Instance
                         .forItems(ExNihiloBlocks.CRUCIBLE_UNFIRED.get()))
                 .build(consumer, createSaveLocation(ExNihiloBlocks.CRUCIBLE_FIRED.getId()));
+        CookingRecipeBuilder
+                .blastingRecipe(Ingredient.fromItems(ExNihiloBlocks.CRUCIBLE_UNFIRED.get()), ExNihiloBlocks.CRUCIBLE_FIRED
+                        .get(), 0.7F, 200)
+                .addCriterion("has_uncooked_crucible", InventoryChangeTrigger.Instance
+                        .forItems(ExNihiloBlocks.CRUCIBLE_UNFIRED.get()))
+                .build(consumer, createSaveLocation(ExNihiloBlocks.CRUCIBLE_FIRED.getId()));
```
Comment (NovaMachina) [inline src/main/java/novamachina/exnihilosequentia/common/datagen/ExNihiloRecipeGenerator.java]: Duplicate recipe name

**Review by NovaMachina — CHANGES_REQUESTED** [review]
This is really close. Duplicate recipe names aren't allowed though.

**Review by NovaMachina — APPROVED** [review]

**Pull request merged**